### ISSUE TITLE
Pub get failed

### ISSUE THREAD

**Issue by thewh1teagle** [thedevsbuddy/tailwind_cli#12]
```shell
flutter pub add -d tailwind_cli  
    Because tailwind_cli <0.2.1 depends on dcli ^1.12.4 and tailwind_cli >=0.2.1 <0.5.5 depends on dcli ^1.31.2, tailwind_cli <0.5.5 requires dcli ^1.12.4.
    And because tailwind_cli >=0.5.5 <0.5.6 depends on dcli ^1.36.2 and tailwind_cli >=0.5.6 depends on dcli ^1.31.2, every version of tailwind_cli requires dcli ^1.12.4.
(1) So, because dcli >=0.51.0 <2.0.0-beta.13 depends on validators2 ^3.0.0 which depends on vin_decoder ^0.2.0-nullsafety, every version of tailwind_cli requires vin_decoder ^0.2.0-nullsafety.

    Because no versions of vin_decoder match >=0.2.0-nullsafety <0.2.0-nullsafety and vin_decoder >=0.2.0-nullsafety <0.2.1-nullsafety depends on http ^0.13.0, vin_decoder >=0.2.0-nullsafety <0.2.1-nullsafety requires http ^0.13.0.
    And because vin_decoder >=0.2.1-nullsafety depends on http ^0.13.3, vin_decoder >=0.2.0-nullsafety requires http ^0.13.0.
    And because every version of tailwind_cli requires vin_decoder ^0.2.0-nullsafety (1), every version of tailwind_cli requires http ^0.13.0.
    So, because app depends on both http ^1.1.0 and tailwind_cli any, version solving failed.
```

```shell
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.13.9, on Microsoft Windows [Version 10.0.22621.2506], locale he-IL)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.5.5)
[√] Android Studio (version 2022.2)
[√] VS Code (version 1.83.1)
[√] Connected device (4 available)
[√] Network resources

• No issues found!
```

**Comment by thedevsbuddy** [comment]
@thewh1teagle 

As i checked by creating a new project and adding `tailwind_cli` as a dependency i am able to install and use the `tailwind_cli` package in my project.

![image](https://github.com/thedevsbuddy/tailwind_cli/assets/72697792/4421689a-648e-4804-8db0-8654e017e0f3)


And here is my flutter version:

![image](https://github.com/thedevsbuddy/tailwind_cli/assets/72697792/28e6328e-fd4f-4418-b12a-81759f487706)


You might getting the error because of some package in your project which is not compatible with `tailwind_cli`.

**Comment by SachinAgarwal1337** [comment]
@thedevsbuddy 1stly This is an amazing package. I liked this better compared to other ones like velocityx and stilo. Thank you so much for this. 

I am also facing similar problem

```
    Because dcli >=1.5.0 <1.31.3 depends on ffi ^1.1.1 and dcli >=1.30.3 <1.33.2 depends on win32 ^2.6.1, dcli >=1.5.0 <1.33.2 requires ffi ^1.1.1 or win32
      ^2.6.1.
    And because dcli >=1.33.2 <1.35.7 depends on win32 ^2.7.0 and dcli >=1.35.7 <2.0.0-beta.5 depends on win32 ^3.0.0, dcli >=1.5.0 <2.0.0-beta.5 requires
      win32 ^2.6.1 or ^3.0.0 or ffi ^1.1.1.
(1) So, because share_plus >=6.0.0 depends on ffi ^2.0.1 and share_plus >=7.0.1 depends on win32 >=4.0.0 <6.0.0, share_plus >=7.0.1 is incompatible with dcli
      >=1.5.0 <2.0.0-beta.5.

    Because tailwind_cli <0.2.1 depends on dcli ^1.12.4 and tailwind_cli >=0.2.1 <0.5.5 depends on dcli ^1.31.2, tailwind_cli <0.5.5 requires dcli ^1.12.4.
    And because tailwind_cli >=0.5.5 <0.5.6 depends on dcli ^1.36.2 and tailwind_cli >=0.5.6 depends on dcli ^1.31.2, every version of tailwind_cli requires
      dcli ^1.12.4.
    And because share_plus >=7.0.1 is incompatible with dcli >=1.5.0 <2.0.0-beta.5 (1), share_plus >=7.0.1 is incompatible with tailwind_cli.
    So, because zoop depends on both share_plus ^7.2.1 and tailwind_cli any, version solving failed.
```

I understand `share_plus` package seems to be causing this issue because of some dependency conflict. But my point is `share_plus` seems to be a very common package which most of the apps might need.
And I think `tailwind_cli` should be compatible with it.
From the looks of it, I think if the packages in `tailwind_cli` are updated, this should fix it. But I can be wrong.
And I dont know if it is supposed to be `tailwind_cli` responsibility to make sure it is compatible or not. But just brining this to your notice in case if it helps.

Again thank you so much for this wonderful Package. 🎉

**Comment by thedevsbuddy** [comment]
@SachinAgarwal1337 

Thanks for your feedback about the project.

And i will look into the issue that you are facing, Currently i am working on the new version, In new version i am also updating the packages it uses. I think it will solve the issue.

**Comment by SachinAgarwal1337** [comment]
@thedevsbuddy That sounds amazing!!! Looking for for it

**Comment by thedevsbuddy** [comment]
@SachinAgarwal1337 Can you please share the Flutter SDK version you are using?

**Comment by SachinAgarwal1337** [comment]
@thedevsbuddy 

```
Flutter 3.16.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 78666c8dc5 (6 days ago) • 2023-12-19 16:14:14 -0800
Engine • revision 3f3e560236
Tools • Dart 3.2.3 • DevTools 2.28.4
```

**Comment by thedevsbuddy** [comment]
@SachinAgarwal1337 Can you please check now as i checked in newly created flutter project it was successfully installed.

As i published a new version with latest updated version of required packages.


```
$ flutter pub add -d tailwind_cli
Resolving dependencies... 
+ archive 3.4.9
+ args 2.4.2
+ chunked_stream 1.4.2
+ circular_buffer 0.11.0
+ completer_ex 4.0.0
+ convert 3.1.1
+ crypto 3.0.3
+ csv 5.1.1
+ dart_console2 3.0.0
+ dcli 3.2.3
+ dcli_core 3.2.3
+ dcli_terminal 0.2.0
+ equatable 2.0.5
+ ffi 2.1.0
+ file 6.1.4 (7.0.0 available)
  flutter_lints 2.0.3 (3.0.1 available)
+ fsm2 3.0.0
+ functional_data 1.1.1
+ glob 2.1.2
+ globbing 1.0.0
+ ini 2.1.0
+ intl 0.18.1 (0.19.0 available)
+ js 0.6.7
+ json2yaml 3.0.1
  lints 2.1.1 (3.0.0 available)
+ lists 1.0.1
+ logger 2.0.2+1
+ logging 1.2.0
  matcher 0.12.16 (0.12.16+1 available)
  material_color_utilities 0.5.0 (0.8.0 available)
  meta 1.10.0 (1.11.0 available)
+ mime 1.0.4
  path 1.8.3 (1.9.0 available)
+ pointycastle 3.7.3
+ posix 6.0.1
+ pub_semver 2.1.4
+ pubspec_lock 3.0.2
+ pubspec_manager 1.0.0-alpha.11
+ scope 4.1.0
+ settings_yaml 7.1.0
+ simple_logger 1.9.0+3
+ sprintf 7.0.0
+ stacktrace_impl 2.3.0
+ strings 2.0.3
+ sum_types 0.3.5
+ synchronized 3.1.0+1
+ system_info2 4.0.0
+ tailwind_cli 0.7.2
  test_api 0.6.1 (0.7.0 available)
+ tree_iterator 3.0.0
+ typed_data 1.3.2
+ unicode 0.3.1
+ uuid 4.2.2 (4.3.1 available)
+ validators2 5.0.0
  web 0.3.0 (0.4.0 available)
+ win32 4.1.4 (5.1.1 available)
+ yaml 3.1.2
Changed 49 dependencies!
12 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
```

My flutter version

```
$ flutter --version
Flutter 3.16.2 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 9e1c857886 (4 weeks ago) • 2023-11-30 11:51:18 -0600
Engine • revision cf7a9d0800
Tools • Dart 3.2.2 • DevTools 2.28.3
```

**Comment by SachinAgarwal1337** [comment]
@thedevsbuddy The new version worked like a charm. Thank you so much for quick response and fix.
Although there was this one small issue, not sure if it is from my end.
when I ran `flutter pub add dev:tailwind_cli` it dint work. It gave me same error.
But when I added the tailwind_cli in `pubspec.yaml` directly and ran `flutter pub get` it worked like a charm

**Comment by thedevsbuddy** [comment]
@SachinAgarwal1337 Actually it was the issue on both side code in this package was having issue according to new flutter SDK and also the `dcli` package was outdated.

**Comment by thedevsbuddy** [comment]
Fixed with 3c03ce77845e995ba9716883d6afb06bda9849fb commit

**Comment by SachinAgarwal1337** [comment]
@thewh1teagle Thank you. Also I wanted to share a suggestion. Could not find the right place to share this so sharing it here.
It will be best if the package is usable without its core widgets also. Like If I can just get the values by calling methods. 

Eg: `TwRounded.sm()` will return `RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(2.0))` 

or 


If it can be used in following way:
 `RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(TwRounded.sm()))`

This will make it more robust. And we will be able to set default theme as well
